src/apispec/ext/marshmallow/openapi.py

```diff
@@ -110,35 +110,29 @@ def resolve_nested_schema(self, schema):
         return self.get_ref_dict(schema_instance)
 
     def schema2parameters(
-        self,
-        schema,
-        *,
-        default_in="body",
-        name="body",
-        required=False,
-        description=None
+        self, schema, *, location, name="body", required=False, description=None
```

Comment:
I'm torn about what to do with the `name` parameter. It's only used when `location=="body"` and it's default value is tied to that case - I presume to guarantee a valid parameter object if the user doesn't enter anything. Seems like it's adding a little extra confusion to this method but I'm not sure anything better can be done.